The version string is missing as a part of `/subscription` response

It seems that the version field is empty in for a subscription: 
When trying to run `test_id_conversion_bug_v5` both the `PUT` and the `GET` gives an empty version string. 

```
=================================== FAILURES ===================================
__________________________ test_id_conversion_bug_v5 ___________________________

ids = <function ids.<locals>.<lambda> at 0x7f857df8a430>, scd_api = '0.3.17'
scd_session = <monitoring.monitorlib.infrastructure.DSSTestSession object at 0x7f857e0af280>

    @for_api_versions(scd.API_0_3_5, scd.API_0_3_17)
    @default_scope(SCOPE_SC)
    def test_id_conversion_bug_v5(ids, scd_api, scd_session):
      sub_uuid = ids(SUB_TYPE)
      time_ref = datetime.datetime.utcnow() + datetime.timedelta(days=1)
      time_start = datetime.datetime(time_ref.year, time_ref.month, time_ref.day, 1, 30)
      time_end = datetime.datetime(time_ref.year, time_ref.month, time_ref.day, 22, 15)
      req = {# ...removed for brevity...
        "old_version": 0,
       # ...removed for brevity...
      }
      resp = scd_session.put('/subscriptions/{}'.format(sub_uuid), json=req)
      assert resp.status_code == 200, resp.content
      data = resp.json()['subscription']
      print("PUT /subscription/{s_id}\n", "\n".join("{} -> '{}'".format(i, data[i]) for i in data))

      req["extents"]["time_start"]["value"] = (time_start + datetime.timedelta(hours=1)).isoformat() + "Z"
      req["old_version"] = 1

      resp = scd_session.get('/subscriptions/{}'.format(sub_uuid), json=req)
      assert resp.status_code == 200, resp.content
      data = resp.json()['subscription']
      print("GET /subscription/{s_id}\n", "\n".join("{} -> '{}'".format(i, data[i]) for i in data))

      resp = scd_session.put('/subscriptions/{}/{}'.format(sub_uuid, resp.json()['subscription']['version']), json=req)
>     assert resp.status_code == 200, resp.content
E     AssertionError: b'{
E         "error": "Missing version string",
E         "code": 3,
E         "message": "Missing version string",
E         "error_id": "E:7d2f98d9-bca5-42b7-aa15-1c67d7168bcd"
E       }'
E     assert 400 == 200
E      +  where 400 = <Response [400]>.status_code

scd/test_operation_special_cases.py:202: AssertionError
----------------------------- Captured stdout call -----------------------------
PUT /subscription/{s_id}
 dependent_operational_intents -> '[]'
id -> '000000d4-ffff-40ff-a64b-a8fb8db60000'
implicit_subscription -> 'False'
notification_index -> '0'
...removed for brevity
version -> ''

GET /subscription/{s_id}
 dependent_operational_intents -> '[]'
id -> '000000d4-ffff-40ff-a64b-a8fb8db60000'
...removed for brevity
version -> ''
```

--- a/build/deploy/db_schemas/scd/000002_support_api_1_0_0.up.sql
+++ b/build/deploy/db_schemas/scd/000002_support_api_1_0_0.up.sql
@@ -1,6 +1,5 @@
-/* Switch Subscription version from integer to string */
-SET enable_experimental_alter_column_type_general = true;
-ALTER TABLE scd_subscriptions ALTER COLUMN version SET DATA TYPE STRING;
+/* Note: Subscription version column is now ignored; version, like OVN for
+   operational intent, is encoded in updated_at */
 
 /* Add tracking for operational intent state */
 CREATE TYPE operational_intent_state AS ENUM ('Unknown', 'Accepted', 'Activated', 'Nonconforming', 'Contingent');

--- a/pkg/scd/models/subscriptions.go
+++ b/pkg/scd/models/subscriptions.go
@@ -24,8 +24,12 @@ const (
 
 // Subscription represents an SCD subscription
 type Subscription struct {
-	ID                          dssmodels.ID
-	Version                     *dssmodels.Version
+	ID dssmodels.ID
+
+	// Version is an OVN-like string constructed from the Subscription's
+	// updated_at field in the database; it may be unspecified when creating a new
+	// Subscription in the database.
+	Version                     OVN
 	NotificationIndex           int
 	Manager                     dssmodels.Manager
 	StartTime                   *time.Time

--- a/pkg/scd/operational_intents_handler.go
+++ b/pkg/scd/operational_intents_handler.go
@@ -355,7 +355,6 @@ func (a *Server) PutOperationalIntentReference(ctx context.Context, entityid str
 			sub, err = r.UpsertSubscription(ctx, &scdmodels.Subscription{
 				ID:                          dssmodels.ID(uuid.New().String()),
 				Manager:                     manager,
-				Version:                     dssmodels.VersionFromTime(time.Now()),
 				StartTime:                   uExtent.StartTime,
 				EndTime:                     uExtent.EndTime,
 				AltitudeLo:                  uExtent.SpatialVolume.AltitudeLo,

--- a/pkg/scd/store/cockroach/subscriptions.go
+++ b/pkg/scd/store/cockroach/subscriptions.go
@@ -96,7 +96,7 @@ func (c *repo) fetchSubscriptions(ctx context.Context, q dsssql.Queryable, query
 		var (
 			s         = new(scdmodels.Subscription)
 			updatedAt time.Time
-			version   string
+			version   int
 		)
 		err = rows.Scan(
 			&s.ID,
@@ -111,10 +111,12 @@ func (c *repo) fetchSubscriptions(ctx context.Context, q dsssql.Queryable, query
 			&s.EndTime,
 			&updatedAt,
 		)
-		v, _ := dssmodels.VersionFromString(version)
-		s.Version = v
 		if err != nil {
 			return nil, stacktrace.Propagate(err, "Error scanning Subscription row")
+		}
+		s.Version = scdmodels.NewOVNFromTime(updatedAt, s.ID.String())
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "Error generating Subscription version")
 		}
 		payload = append(payload, s)
 	}
@@ -209,7 +211,7 @@ func (c *repo) pushSubscription(ctx context.Context, q dsssql.Queryable, s *scdm
 	s, err := c.fetchSubscription(ctx, q, upsertQuery,
 		s.ID,
 		s.Manager,
-		s.Version.String(),
+		0,
 		s.USSBaseURL,
 		s.NotificationIndex,
 		s.NotifyForOperationalIntents,

--- a/pkg/scd/subscriptions_handler.go
+++ b/pkg/scd/subscriptions_handler.go
@@ -20,19 +20,16 @@ var (
 )
 
 func (a *Server) CreateSubscription(ctx context.Context, req *scdpb.CreateSubscriptionRequest) (*scdpb.PutSubscriptionResponse, error) {
-	return a.PutSubscription(ctx, req.GetSubscriptionid(), nil, req.GetParams())
+	return a.PutSubscription(ctx, req.GetSubscriptionid(), "", req.GetParams())
 }
 
 func (a *Server) UpdateSubscription(ctx context.Context, req *scdpb.UpdateSubscriptionRequest) (*scdpb.PutSubscriptionResponse, error) {
-	version, err := dssmodels.VersionFromString(req.GetVersion())
-	if err != nil {
-		return nil, stacktrace.PropagateWithCode(err, dsserr.BadRequest, "Failed to parse version")
-	}
+	version := req.GetVersion()
 	return a.PutSubscription(ctx, req.GetSubscriptionid(), version, req.GetParams())
 }
 
 // PutSubscription creates a single subscription.
-func (a *Server) PutSubscription(ctx context.Context, subscriptionid string, version *dssmodels.Version, params *scdpb.PutSubscriptionParameters) (*scdpb.PutSubscriptionResponse, error) {
+func (a *Server) PutSubscription(ctx context.Context, subscriptionid string, version string, params *scdpb.PutSubscriptionParameters) (*scdpb.PutSubscriptionResponse, error) {
 	// Retrieve Subscription ID
 	id, err := dssmodels.IDFromString(subscriptionid)
 
@@ -71,7 +68,7 @@ func (a *Server) PutSubscription(ctx context.Context, subscriptionid string, ver
 	subreq := &scdmodels.Subscription{
 		ID:      id,
 		Manager: manager,
-		Version: version,
+		Version: scdmodels.OVN(version),
 
 		StartTime:  extents.StartTime,
 		EndTime:    extents.EndTime,
@@ -108,17 +105,17 @@ func (a *Server) PutSubscription(ctx context.Context, subscriptionid string, ver
 
 		if old == nil {
 			// There is no previous Subscription (this is a creation attempt)
-			if !subreq.Version.Empty() {
+			if subreq.Version.String() != "" {
 				// The user wants to update an existing Subscription, but one wasn't found.
 				return stacktrace.NewErrorWithCode(dsserr.NotFound, "Subscription %s not found", subreq.ID.String())
 			}
 		} else {
 			// There is a previous Subscription (this is an update attempt)
 			switch {
-			case subreq.Version.Empty():
+			case subreq.Version.String() == "":
 				// The user wants to create a new Subscription but it already exists.
 				return stacktrace.NewErrorWithCode(dsserr.AlreadyExists, "Subscription %s already exists", subreq.ID.String())
-			case !subreq.Version.Matches(old.Version):
+			case subreq.Version.String() != old.Version.String():
 				// The user wants to update a Subscription but the version doesn't match.
 				return stacktrace.Propagate(
 					stacktrace.NewErrorWithCode(dsserr.VersionMismatch, "Subscription version %s is not current", subreq.Version),

